src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,34 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
+import java.util.ArrayList;
+import java.util.List;
+
 public class RegistrationServiceImpl implements RegistrationService {
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws ImpossibleRegisterUserException {
+        if (user == null) {
+            throw new ImpossibleRegisterUserException("The user can`t be empty");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new ImpossibleRegisterUserException("This login is incorrect");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new ImpossibleRegisterUserException("This password is incorrect");
+        } else if (user.getAge() == null || user.getAge() < 18) {
```

Comment:
The age validation checks if the age is null or less than 18. Ensure this aligns with the task requirements. If the task specifies a different age requirement, modify this check accordingly.